e2e/playwright/tests/folder-rules/page-objects/components/conditions.component.ts

```diff
@@ -29,22 +29,33 @@ export class ConditionComponent extends ManageRulesDialogComponent {
     super(page);
   }
 
-  private async selectField(fields: Partial<Field>): Promise<void> {
-    await this.fieldDropDown.click();
+  private async selectField(fields: Partial<Field>, index: number): Promise<void> {
+    await this.fieldDropDown.nth(index).click();
     const option = this.getOptionLocator(fields);
     await option.click();
   }
 
-  private async selectComparator(comparators: Partial<Comparator>): Promise<void> {
-    await this.comparatorDropDown.click();
+  private async selectComparator(comparators: Partial<Comparator>, index: number): Promise<void> {
+    await this.comparatorDropDown.nth(index).click();
     const option = this.getOptionLocator(comparators);
     await option.click();
   }
 
-  async addCondition(fields: Partial<Field>, comparators: Partial<Comparator>, value: string): Promise<void> {
+  async addCondition(fields: Partial<Field>, comparators: Partial<Comparator>, value: string, index: number): Promise<void> {
     await this.addConditionButton.click();
-    await this.selectField(fields);
-    await this.selectComparator(comparators);
-    await this.typeConditionValue(value);
+    await this.selectField(fields, index);
+    await this.selectComparator(comparators, index);
+    await this.typeConditionValue(value, index);
+  }
+
+  async createConditionGroup(): Promise<void> {
```

Comment:
We shouldn't have separate methods for one line button clicks

Reply:
This method specific to create new Condition group as we do not create it for each new condition added, but I think  if we can add this step to 'AddConditionGroup' method.

Reply:
I keep this method as is, we might add more steps in this when we try to add more logic to create more complex groups and condition structure.